<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OptionsDialog</class>
 <widget class="QDialog" name="OptionsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>440</width>
    <height>400</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>440</width>
    <height>400</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>440</width>
    <height>400</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>FFmpeg Path</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="verticalSpacing">
       <number>0</number>
      </property>
      <item row="0" column="0" alignment="Qt::AlignmentFlag::AlignLeft">
       <widget class="QRadioButton" name="rbFfAuto">
        <property name="text">
         <string>Find Automatically</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">bgpSoundfontPath</string>
        </attribute>
       </widget>
      </item>
      <item row="0" column="2" rowspan="2" alignment="Qt::AlignmentFlag::AlignTop">
       <widget class="QLabel" name="lbFfmpegFound">
        <property name="styleSheet">
         <string notr="true">QLabel {
	color: #00cc00;
}</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="3">
       <widget class="controls::PathChooser" name="pcFfmpegPath" native="true">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>5</height>
         </size>
        </property>
        <property name="focusPolicy">
         <enum>Qt::FocusPolicy::TabFocus</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="rbFfManual">
        <property name="text">
         <string>Use Path:</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">bgpSoundfontPath</string>
        </attribute>
       </widget>
      </item>
      <item row="0" column="1" rowspan="2">
       <widget class="QLabel" name="lbFfmpegNotFound">
        <property name="styleSheet">
         <string notr="true">QLabel {
color: red;
}</string>
        </property>
        <property name="text">
         <string>Could not find FFmpeg in the system path!</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Soundfont Path</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="controls::PathChooser" name="pcSoundfontPath" native="true">
        <property name="focusPolicy">
         <enum>Qt::FocusPolicy::TabFocus</enum>
        </property>
        <property name="filter" stdset="0">
         <string>Soundfonts (*.sf2 *.sfz)</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Encoding Options</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,0,1">
      <property name="horizontalSpacing">
       <number>12</number>
      </property>
      <item row="1" column="0">
       <widget class="QWidget" name="widget" native="true">
        <layout class="QFormLayout" name="formLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Codec</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="cmbVideoCodec">
           <item>
            <property name="text">
             <string>H.264</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>H.265 (HEVC)</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_4">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Encoding speed (aka the ffmpeg &amp;quot;-preset&amp;quot; flag). Higher speeds generally correlate with less efficient encoding/larger file sizes.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Speed&lt;span style=&quot; text-decoration: underline; vertical-align:super;&quot;&gt;?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="cmbEncodeSpeed">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <item>
            <property name="text">
             <string>Ultra Fast</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Super Fast</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Very Fast</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Faster</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Fast</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Medium</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Slow</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Slower</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Very Slow</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Short for &amp;quot;constant rate factor.&amp;quot; Lower values result in higher video quality but larger file sizes. Should generally be set between 17 and 32ish. The default value is 23 for H.264 and 28 for H.265.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;CRF&lt;span style=&quot; text-decoration: underline; vertical-align:super;&quot;&gt;?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="textFormat">
            <enum>Qt::TextFormat::RichText</enum>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="sbCrf">
           <property name="maximum">
            <number>51</number>
           </property>
           <property name="value">
            <number>23</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>&lt;b&gt;Video&lt;/b&gt;</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>&lt;b&gt;Audio&lt;/b&gt;</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QWidget" name="widget_2" native="true">
        <layout class="QFormLayout" name="formLayout_2">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Bitrate</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="sbAudioBitrate">
           <property name="suffix">
            <string> kbps</string>
           </property>
           <property name="minimum">
            <number>128</number>
           </property>
           <property name="maximum">
            <number>256</number>
           </property>
           <property name="value">
            <number>192</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="1" rowspan="2">
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Orientation::Vertical</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Save</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>controls::PathChooser</class>
   <extends>QWidget</extends>
   <header>src/controls/pathchooser.h</header>
   <container>1</container>
   <slots>
    <signal>path_changed(QString)</signal>
    <slot>open_file_chooser()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>rbFfAuto</tabstop>
  <tabstop>rbFfManual</tabstop>
  <tabstop>pcFfmpegPath</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>OptionsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>266</x>
     <y>390</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>OptionsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>334</x>
     <y>390</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rbFfAuto</sender>
   <signal>toggled(bool)</signal>
   <receiver>OptionsDialog</receiver>
   <slot>enable_ffmpeg_check_timer(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>63</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rbFfManual</sender>
   <signal>toggled(bool)</signal>
   <receiver>pcFfmpegPath</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>69</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>80</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbVideoCodec</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>OptionsDialog</receiver>
   <slot>reset_crf_to_default()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>134</x>
     <y>233</y>
    </hint>
    <hint type="destinationlabel">
     <x>219</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>enable_ffmpeg_check_timer(bool)</slot>
  <slot>reset_crf_to_default()</slot>
 </slots>
 <buttongroups>
  <buttongroup name="bgpSoundfontPath"/>
 </buttongroups>
</ui>
